backend/src/main/java/com/bakdata/conquery/models/events/stores/specific/DateRangeTypeCompound.java

```diff
@@ -0,0 +1,140 @@
+package com.bakdata.conquery.models.events.stores.specific;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import com.bakdata.conquery.io.cps.CPSType;
+import com.bakdata.conquery.models.common.daterange.CDateRange;
+import com.bakdata.conquery.models.events.Bucket;
+import com.bakdata.conquery.models.events.stores.primitive.IntegerDateStore;
+import com.bakdata.conquery.models.events.stores.root.ColumnStore;
+import com.bakdata.conquery.models.events.stores.root.DateRangeStore;
+import com.bakdata.conquery.models.events.stores.root.DateStore;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.ToString;
+
+/**
+ * Stores {@link CDateRange} by storing only the references of the start and end of the {@link CDateRange}.
+ * After the deserialisation, stores are instantiated as {@link DateStore}s using the references to the columns.
+ */
+@CPSType(base = ColumnStore.class, id = "DATE_RANGE_COMPOUND")
+@Getter
+@Setter
+@ToString(of = {"startColumn", "endColumn"})
+public class DateRangeTypeCompound implements DateRangeStore {
+
+
+	@NotNull
+	@NotEmpty
+	private String startColumn, endColumn;
+
+	// does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	@JsonIgnore
+	private DateStore startStore;
+
+	// does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	@JsonIgnore
+	private DateStore endStore;
+
+
+	/**
+	 * Represents the bucket from where this will be saved
+	 * This bucket will be set after the deserialization
+	 */
+	@Setter(AccessLevel.PROTECTED)
+	@JsonIgnore
+	private Bucket parent;
+
+	@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+	public DateRangeTypeCompound(String startColumn, String endColumn) {
+		setStartColumn(startColumn);
+		setEndColumn(endColumn);
+	}
+
+	@JsonIgnore
+	public DateStore getStartStore() {
+		if (getParent() == null) {
+			return null;
+		}
+		if (startStore == null) {
+			setStartStore((DateStore) getParent().getStore(getStartColumn()));
+		}
+		return startStore;
+	}
+
+	@JsonIgnore
+	public DateStore getEndStore() {
+		if (getParent() == null) {
+			return null;
+		}
+		if (endStore == null) {
+			setEndStore((DateStore) getParent().getStore(getEndColumn()));
+		}
+		return endStore;
+	}
+
+	@Override
+	public void setParent(@NonNull Bucket bucket) {
+		parent = bucket;
+	}
+
+	@Override
+	public int getLines() {
+		if (getStartStore() == null && getEndStore() == null) {
+			return 0;
+		}
+		if (getStartStore() == null || getEndStore() == null) {
+			throw new IllegalStateException(String.format("DateRangeTypeCompound[%s] has one null store.", this));
+		}
```

Comment:
Das kann denke ich nicht passieren oder, da dur ja eine Exception in getStoreByName oder Table wirfst? Bzw die Exception sollte auf jeden Fall an der Stelle schon geworfen worden sein und nicht hier.